assembly/src/assembler/mod.rs

```diff
@@ -116,10 +120,14 @@ impl Assembler {
         Self::default()
     }
 
-    /// Start building an [Assembler] with the given [Kernel].
-    pub fn with_kernel(kernel: Kernel) -> Self {
+    /// Start building an [`Assembler`] with the given [`Kernel`] and the [`MastForest`] that was
+    /// used to compile the kernel.
+    pub fn with_kernel(kernel: Kernel, mast_forest: MastForest) -> Self {
```

Comment:
I think the need to pass in `mast_forest` for the kernel will go away in the next PR.